homeassistant/components/sensor/folder.py

```diff
@@ -13,27 +14,49 @@
 
 from homeassistant.helpers.entity import Entity
 import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
 from homeassistant.components.sensor import PLATFORM_SCHEMA
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_FOLDER_PATHS = 'folder'
+CONF_FOLDER_PATH = 'folder'
 CONF_FILTER = 'filter'
+CONF_RECURSIVE = 'recursive'
 DEFAULT_FILTER = '*'
+DEFAULT_NAME = ''
+DEFAULT_RECURSIVE = False
+FILE = 'file'
+SIGNAL_FILE_ADDED = 'file_added'
+SIGNAL_FILE_DELETED = 'file_deleted'
+SIGNAL_FILE_MODIFIED = 'file_modified'
 
 SCAN_INTERVAL = timedelta(seconds=1)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_FOLDER_PATHS): cv.isdir,
+    vol.Required(CONF_FOLDER_PATH): cv.isdir,
     vol.Optional(CONF_FILTER, default=DEFAULT_FILTER): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_RECURSIVE, default=DEFAULT_RECURSIVE): cv.boolean,
 })
 
 
-def get_files_list(folder_path, filter_term):
-    """Return the list of files, applying filter."""
-    query = folder_path + filter_term
-    files_list = glob.glob(query)
-    return files_list
+def get_timestamp(file_path):
+    """Return the timestamp of file."""
+    mtime = os.stat(file_path).st_mtime
```

Comment:
you're again doing multiple stats for each file. You're putting a lot of strain on the file system